webob/response.py

```diff
@@ -1287,13 +1296,11 @@ def __call__(self, environ, start_response):
         if self.conditional_response:
             return self.conditional_response_app(environ, start_response)
 
-        # Build headerlist, if there is a Location header, make it absolute
-        headerlist = [
-            (k, v) if k.lower() != 'location'
-            else (k, self._make_location_absolute(environ, v))
-            for (k, v)
-            in self._headerlist
-        ]
+        # If the HTTP Status code might have a Location header, try to make it absolute
+        if self._status[:3] in ('201', '301', '302', '303', '305', '307'):
```

Comment:
Setting Location on a `200 OK` HTTP Status code is not allowed for example, and will cause some intermediaries to change the status code to a 302 (Apache's mod_proxy IIRC), and others will send a 500 Server Error back to the remote client (Squid). Various HTTP checkers (for example OpenStack has some neat ones) also complain loudly. You can't just set Location on stuff willy nilly. Same thing with Content-Length for example, you can't set that on responses without a body.

I can add 308 to the list, or I can drop the optimization completely. It doesn't make that big of a difference.